feat: add sockProtocol option to overlay (#242)
pmmmwh/react-refresh-webpack-plugin: docs/API.md

  sockHost?: string;
  sockPath?: string;
  sockPort?: number;

>   sockProtocol?: 'http' | 'https';

Reviewer: I did a quick search on GitHub and I think `ws` and `wss` protocols are getting more traction - we should also allow them here (please update all other places related to this enum too!)

Set a custom path for the error overlay to listen to Webpack build messages.
Useful if you set `devServer.sockPath` to something other than `/sockjs-node`.

### `sockProtocol`

Default: parsed from current url

> Type: `enum`

Reviewer: I think it would be better for use to be explicit about this - what about:

```suggestion
Type: `http:` or `https:`
```

(refernce `sockIntegration`)

Reviewer: As a note to that, I think `ws` and `wss` are also technically valid choices if people use strict, native WebSockets (not `SockJS`)

Reviewer: Another note I think would be relavent - I'm more inclined to have `http` instead of `http:` here. I think we should either go the full way (`http://`) or just go with the protocol name (`http`).

Author: Yup, I wanted to do that, just wasn’t sure. Updated

**This is relevant for `webpack-dev-server` only.**

> Force protocol type for the error overlay to listen to Webpack build messages.
> Useful if you want to force https communication.

Reviewer: ```suggestion
Force a protocol for the error overlay to listen for Webpack build messages.
Useful if you want to enforce https communication, or if you're working under a non-HTTP path.
```

Default: parsed from current url

> Type: `http`, `https`, `ws`, `wss`

Reviewer: ```suggestion
Type: `http`, `https`, `ws` or `wss`
```
